app/src/main/java/com/fmsys/snapdrop/SettingsFragment.java

```diff
@@ -146,6 +146,18 @@ public boolean onExtraClicked(final @NonNull View view, final @NonNull SpecialBu
             });
         }
 
+        final Preference showConnectivityCardPref = findPreference(getString(R.string.pref_show_connectivity_card));
+
+            showConnectivityCardPref.setVisible(true);
+            showConnectivityCardPref.setOnPreferenceChangeListener((pref, newValue) -> {
+
+                PreferenceManager.getDefaultSharedPreferences(getContext()).edit().putBoolean(showConnectivityCardPref.getKey(), (boolean) newValue).apply();
+                return true;
+            });
+
+
+
+
```

Comment:
This whole code part is not necessary, the android `Preferences` framework will automatically store preference changes for you ;-)